provider/techstack/index.ts

```diff
@@ -0,0 +1,94 @@
+
+import fs from 'fs'
+import path from 'path'
+import YAML from 'yaml'
+import {
+    type AnnotationsParams,
+    type AnnotationsResult,
+    type CapabilitiesParams,
+    type CapabilitiesResult,
+    type Item,
+    type Provider
+} from '@opencodegraph/provider'
+
+import type TSF from './techstack.schema'
+export interface Settings { yaml: string }
+
+/**
+ * Read the techstack file configuration for project
+ *
+ * @param filename - techstack yml filename
+ * @returns parsed yaml object
+ */
+async function load(filename: string): Promise<TSF> {
```

Comment:
Being able to automatically infer the techstack.yml file seems quite important here. Unfortunately, OpenCodeGraph does not yet have a way to help you do this. I was thinking that providers could say in their `capabilities` response something like "please also send me the contents of the following files with each annotations request: techstack.yml, ...". What do you think about that approach?

Reply:
I'm not sure I entirely follow you here but here are my thoughts.

I didn't intend to automatically infer the techstack.yml file - it's set explicitly in the provider settings and the `load()` method simply references that in L59 and I find the explicitness in this design better.
```typescript
const report = await load(settings.yaml)
```

That said, it's maybe not impossible to automatically infer a techstack.yml file for a respository and supplement it with an appropriate override in provider settings. You could also introduce an optional member in the `AnnotationsParams` interface that sends any techstack.yml data for each annotation request.

However, I don't see any real gain in these approaches vs the current settings driven design. Just my 2 cents.

Reply:
Fyi, I've dropped all path resolutions and adhering reading techstack.yml content from an absolute path provided in the provider settings. I'll document this in a README, see the below commits.

- https://github.com/sourcegraph/opencodegraph/pull/12/commits/25c128dd34ea52d92eae82872f3a8f101141d4fb
- https://github.com/sourcegraph/opencodegraph/pull/12/commits/6757344101fe385a778487f900461e2b3fdf2ed7